### ISSUE TITLE
[Feature]: Add support for levels and experience points (XP)

### ISSUE THREAD

**Issue by lulunac27a** [dromse/obsidian-gamified-tasks#98]
### What kind of feature do you want?

I want to see my level with experience points (XP) to see my progress so far.

### Additional questions

- [ ] I want to work on this feature request

**Comment by dromse** [comment]
Hi, @lulunac27a! I'm pondering how to incorporate this feature inside the plugin. 
I consider a few variants for earning points (could be implemented all or pick the most useful):
1. Abstract Points
2. Points based on counter
3. Points based on coin earning
4. Use `#lvl/points` to define how many points you gain

Levels:
1. Abstract levels like 1-99 etc.
2. User-specified levels you can define your Level or Rank in the settings.

It is currently in the design stage, so if you have more detailed thoughts on implementation, I will try to implement it.

I'm looking forward to your feedback!

Thanks for your attention.

**Comment by lulunac27a** [comment]
I want to add support with unlimited levels, which has no level cap.

**Comment by dromse** [comment]
Okay, what do you think about using difficulty as an XP and increase the level-up costs geometrically? For instance, the cost could be 5 coins for level 1, 10 coins for level 2, 20 coins for level 3, just multiply on 2 and so on.

**Comment by lulunac27a** [comment]
When the task is completed, the user will get XP points based on the task's difficulty. The more difficult the task, the more XP points the user gets. Leveling up requires a specific amount of XP points to reach the level, so it doesn't cost anything to level up; it levels up automatically when you reach a specified amount of XP points required to level up.

**Comment by dromse** [comment]
Yup, I just use a term coin to describe difficulty, but yeah XP and coins would be independent from each other.

**Comment by lulunac27a** [comment]
I agree with that.

**Comment by lulunac27a** [comment]
Also add a progress bar where the progress is the number of XP since the user levelled up out of the number of XP required to reach the next level.

**Comment by lulunac27a** [comment]
The user can also earn XP points using counters, so when the user click the + sign to increase the counter, the user earns XP points, but when the user click the - (minus) sign, the user don't lose any XP points.

**Comment by dromse** [comment]
What to do if a user miss-clicks the counter and wants to fix it by decreasing it?

**Comment by lulunac27a** [comment]
The user doesn't lose any XP points so the user doesn't lose a level.

**Comment by dromse** [comment]
Okay 👍️

**Comment by lulunac27a** [comment]
Is this work done yet?

**Comment by dromse** [comment]
Hi, @lulunac27a! I apologize but I don't have enough time to work on this project currently, it's planned but in future releases and when I have enough free time to work.

**Comment by lulunac27a** [comment]
Is it almost done with this work?

**Comment by lulunac27a** [comment]
Is the work done? It's taking few months to complete the work.

**Comment by lulunac27a** [comment]
Can you start on this work again? I want to earn XP points and level up.
